### PR TITLE
Fix secret options in windows bootstrap

### LINKED ISSUE #9018: Knife bootstrap is not uploading the secret file specified by `--secret-file` on Windows
## Description
We have recently upgraded to using version 15.3.14, from version 14 and have found that when bootstrapping a windows node the secret file is not being uploaded when specified using the `--secret-file <PATH TO FILE>` parameter, but supplying the raw contents using the `--secret <SECRET>` parameter it is. 

Confirmed the secret file does exist in the location specified by the `--secret-file <PATH TO FILE>` parameter. 

## Chef Version
15.3.14

## Platform Version
windows 10.0.14393

## Replication Case
Doesn't work (doesn't upload the secret file to C:\chef\encrypted_data_bag_secret):
`knife bootstrap winrm://<IP ADDRESS> -U 'USER' -P 'PASS' -E PROD -y -N NODE_NAME -F json --secret-file '<PATH TO SECRET FILE>' --node-ssl-verify-mode none --bootstrap-version 15.3.14 --chef-license accept`

Does work (file is created and contents copied):
`knife bootstrap winrm://<IP ADDRESS> -U 'USER' -P 'PASS' -E PROD -y -N NODE_NAME -F json --secret '<SECRET>' --node-ssl-verify-mode none --bootstrap-version 15.3.14 --chef-license accept`

## Client Output

```
C:/opscode/chef/embedded/lib/ruby/gems/2.6.0/gems/chef-15.3.14-universal-mingw32/lib/chef/encrypted_data_bag_item.rb:134:in `load_secret'
C:/opscode/chef/embedded/lib/ruby/gems/2.6.0/gems/chef-15.3.14-universal-mingw32/lib/chef/dsl/data_query.rb:69:in `data_bag_item'
c:/chef/cache/cookbooks/custom_cookbook1/recipes/default.rb:26:in `from_file'
C:/opscode/chef/embedded/lib/ruby/gems/2.6.0/gems/chef-15.3.14-universal-mingw32/lib/chef/mixin/from_file.rb:34:in `instance_eval'
C:/opscode/chef/embedded/lib/ruby/gems/2.6.0/gems/chef-15.3.14-universal-mingw32/lib/chef/mixin/from_file.rb:34:in `from_file'
C:/opscode/chef/embedded/lib/ruby/gems/2.6.0/gems/chef-15.3.14-universal-mingw32/lib/chef/cookbook_version.rb:199:in `load_recipe'
C:/opscode/chef/embedded/lib/ruby/gems/2.6.0/gems/chef-15.3.14-universal-mingw32/lib/chef/run_context.rb:393:in `load_recipe'
C:/opscode/chef/embedded/lib/ruby/gems/2.6.0/gems/chef-15.3.14-universal-mingw32/lib/chef/run_context.rb:349:in `block in include_recipe'
C:/opscode/chef/embedded/lib/ruby/gems/2.6.0/gems/chef-15.3.14-universal-mingw32/lib/chef/run_context.rb:348:in `each'
C:/opscode/chef/embedded/lib/ruby/gems/2.6.0/gems/chef-15.3.14-universal-mingw32/lib/chef/run_context.rb:348:in `include_recipe'
C:/opscode/chef/embedded/lib/ruby/gems/2.6.0/gems/chef-15.3.14-universal-mingw32/lib/chef/dsl/include_recipe.rb:26:in `include_recipe'
c:/chef/cache/cookbooks/custom_cookbook2/recipes/windows.rb:23:in `from_file'
C:/opscode/chef/embedded/lib/ruby/gems/2.6.0/gems/chef-15.3.14-universal-mingw32/lib/chef/mixin/from_file.rb:34:in `instance_eval'
C:/opscode/chef/embedded/lib/ruby/gems/2.6.0/gems/chef-15.3.14-universal-mingw32/lib/chef/mixin/from_file.rb:34:in `from_file'
C:/opscode/chef/embedded/lib/ruby/gems/2.6.0/gems/chef-15.3.14-universal-mingw32/lib/chef/cookbook_version.rb:199:in `load_recipe'
C:/opscode/chef/embedded/lib/ruby/gems/2.6.0/gems/chef-15.3.14-universal-mingw32/lib/chef/run_context.rb:393:in `load_recipe'
C:/opscode/chef/embedded/lib/ruby/gems/2.6.0/gems/chef-15.3.14-universal-mingw32/lib/chef/run_context.rb:349:in `block in include_recipe'
C:/opscode/chef/embedded/lib/ruby/gems/2.6.0/gems/chef-15.3.14-universal-mingw32/lib/chef/run_context.rb:348:in `each'
C:/opscode/chef/embedded/lib/ruby/gems/2.6.0/gems/chef-15.3.14-universal-mingw32/lib/chef/run_context.rb:348:in `include_recipe'
C:/opscode/chef/embedded/lib/ruby/gems/2.6.0/gems/chef-15.3.14-universal-mingw32/lib/chef/dsl/include_recipe.rb:26:in `include_recipe'
c:/chef/cache/cookbooks/custom_cookbook2/recipes/default.rb:15:in `from_file'
C:/opscode/chef/embedded/lib/ruby/gems/2.6.0/gems/chef-15.3.14-universal-mingw32/lib/chef/mixin/from_file.rb:34:in `instance_eval'
C:/opscode/chef/embedded/lib/ruby/gems/2.6.0/gems/chef-15.3.14-universal-mingw32/lib/chef/mixin/from_file.rb:34:in `from_file'
C:/opscode/chef/embedded/lib/ruby/gems/2.6.0/gems/chef-15.3.14-universal-mingw32/lib/chef/cookbook_version.rb:199:in `load_recipe'
C:/opscode/chef/embedded/lib/ruby/gems/2.6.0/gems/chef-15.3.14-universal-mingw32/lib/chef/run_context.rb:393:in `load_recipe'
C:/opscode/chef/embedded/lib/ruby/gems/2.6.0/gems/chef-15.3.14-universal-mingw32/lib/chef/run_context.rb:349:in `block in include_recipe'
C:/opscode/chef/embedded/lib/ruby/gems/2.6.0/gems/chef-15.3.14-universal-mingw32/lib/chef/run_context.rb:348:in `each'
C:/opscode/chef/embedded/lib/ruby/gems/2.6.0/gems/chef-15.3.14-universal-mingw32/lib/chef/run_context.rb:348:in `include_recipe'
C:/opscode/chef/embedded/lib/ruby/gems/2.6.0/gems/chef-15.3.14-universal-mingw32/lib/chef/dsl/include_recipe.rb:26:in `include_recipe'
c:/chef/cache/cookbooks/custom_cookbook3/recipes/default.rb:23:in `from_file'
C:/opscode/chef/embedded/lib/ruby/gems/2.6.0/gems/chef-15.3.14-universal-mingw32/lib/chef/mixin/from_file.rb:34:in `instance_eval'
C:/opscode/chef/embedded/lib/ruby/gems/2.6.0/gems/chef-15.3.14-universal-mingw32/lib/chef/mixin/from_file.rb:34:in `from_file'
C:/opscode/chef/embedded/lib/ruby/gems/2.6.0/gems/chef-15.3.14-universal-mingw32/lib/chef/cookbook_version.rb:199:in `load_recipe'
C:/opscode/chef/embedded/lib/ruby/gems/2.6.0/gems/chef-15.3.14-universal-mingw32/lib/chef/run_context.rb:393:in `load_recipe'
C:/opscode/chef/embedded/lib/ruby/gems/2.6.0/gems/chef-15.3.14-universal-mingw32/lib/chef/run_context.rb:349:in `block in include_recipe'
C:/opscode/chef/embedded/lib/ruby/gems/2.6.0/gems/chef-15.3.14-universal-mingw32/lib/chef/run_context.rb:348:in `each'
C:/opscode/chef/embedded/lib/ruby/gems/2.6.0/gems/chef-15.3.14-universal-mingw32/lib/chef/run_context.rb:348:in `include_recipe'
C:/opscode/chef/embedded/lib/ruby/gems/2.6.0/gems/chef-15.3.14-universal-mingw32/lib/chef/dsl/include_recipe.rb:26:in `include_recipe'
c:/chef/cache/cookbooks/custom_cookbook4/recipes/default.rb:8:in `from_file'
C:/opscode/chef/embedded/lib/ruby/gems/2.6.0/gems/chef-15.3.14-universal-mingw32/lib/chef/mixin/from_file.rb:34:in `instance_eval'
C:/opscode/chef/embedded/lib/ruby/gems/2.6.0/gems/chef-15.3.14-universal-mingw32/lib/chef/mixin/from_file.rb:34:in `from_file'
C:/opscode/chef/embedded/lib/ruby/gems/2.6.0/gems/chef-15.3.14-universal-mingw32/lib/chef/cookbook_version.rb:199:in `load_recipe'
C:/opscode/chef/embedded/lib/ruby/gems/2.6.0/gems/chef-15.3.14-universal-mingw32/lib/chef/run_context.rb:393:in `load_recipe'
C:/opscode/chef/embedded/lib/ruby/gems/2.6.0/gems/chef-15.3.14-universal-mingw32/lib/chef/run_context/cookbook_compiler.rb:166:in `block in compile_recipes'
C:/opscode/chef/embedded/lib/ruby/gems/2.6.0/gems/chef-15.3.14-universal-mingw32/lib/chef/run_context/cookbook_compiler.rb:163:in `each'
C:/opscode/chef/embedded/lib/ruby/gems/2.6.0/gems/chef-15.3.14-universal-mingw32/lib/chef/run_context/cookbook_compiler.rb:163:in `compile_recipes'
C:/opscode/chef/embedded/lib/ruby/gems/2.6.0/gems/chef-15.3.14-universal-mingw32/lib/chef/run_context/cookbook_compiler.rb:79:in `compile'
C:/opscode/chef/embedded/lib/ruby/gems/2.6.0/gems/chef-15.3.14-universal-mingw32/lib/chef/run_context.rb:223:in `load'
C:/opscode/chef/embedded/lib/ruby/gems/2.6.0/gems/chef-15.3.14-universal-mingw32/lib/chef/policy_builder/expand_node_object.rb:103:in `setup_run_context'
C:/opscode/chef/embedded/lib/ruby/2.6.0/forwardable.rb:230:in `setup_run_context'
C:/opscode/chef/embedded/lib/ruby/gems/2.6.0/gems/chef-15.3.14-universal-mingw32/lib/chef/client.rb:488:in `setup_run_context'
C:/opscode/chef/embedded/lib/ruby/gems/2.6.0/gems/chef-15.3.14-universal-mingw32/lib/chef/client.rb:285:in `run'
C:/opscode/chef/embedded/lib/ruby/gems/2.6.0/gems/chef-15.3.14-universal-mingw32/lib/chef/application.rb:320:in `run_with_graceful_exit_option'
C:/opscode/chef/embedded/lib/ruby/gems/2.6.0/gems/chef-15.3.14-universal-mingw32/lib/chef/application.rb:296:in `block in run_chef_client'
C:/opscode/chef/embedded/lib/ruby/gems/2.6.0/gems/chef-15.3.14-universal-mingw32/lib/chef/local_mode.rb:42:in `with_server_connectivity'
C:/opscode/chef/embedded/lib/ruby/gems/2.6.0/gems/chef-15.3.14-universal-mingw32/lib/chef/application.rb:279:in `run_chef_client'
C:/opscode/chef/embedded/lib/ruby/gems/2.6.0/gems/chef-15.3.14-universal-mingw32/lib/chef/application/base.rb:330:in `run_application'
C:/opscode/chef/embedded/lib/ruby/gems/2.6.0/gems/chef-15.3.14-universal-mingw32/lib/chef/application.rb:69:in `run'
C:/opscode/chef/embedded/lib/ruby/gems/2.6.0/gems/chef-bin-15.3.14/bin/chef-client:25:in `<top (required)>'
C:/opscode/chef/bin/chef-client:176:in `load'
C:/opscode/chef/bin/chef-client:176:in `<main>'
```

## Stacktrace
No secret specified and no secret found at C:\chef\encrypted_data_bag_secret

### PR DESCRIPTION
**Pull request by mwrock**
fixes #9018.

Signed-off-by: mwrock <matt@mattwrock.com>

### CODE REVIEW

File: lib/chef/knife/core/windows_bootstrap_context.rb [hunk 1]
```diff
@@ -49,8 +49,12 @@ def validation_key
           end
         end
 
+        def encrypted_data_bag_secret
+          escape_and_echo(@secret)
+        end
+
         def secret
```
Comment (tas50) [inline lib/chef/knife/core/windows_bootstrap_context.rb:56]: Could we just make this an accessor?

Comment (mwrock) [reply lib/chef/knife/core/windows_bootstrap_context.rb:56]: It's overloading `encrypted_data_bag_secret` in `bootstrap_context.rb`

Comment (mwrock) [reply lib/chef/knife/core/windows_bootstrap_context.rb:56]: oh nevermind

Comment (mwrock) [reply lib/chef/knife/core/windows_bootstrap_context.rb]: made it an accessor

**Pull request merged**